applications/GeoMechanicsApplication/tests/cpp_tests/test_dof_utilities.cpp

```diff
@@ -169,6 +193,110 @@ KRATOS_TEST_CASE_IN_SUITE(VariableTypeAndNodeIDsMustMatchWhenExtractingDofsFromN
+{
+    auto       model           = Model{};
+    const auto nodal_variables = Geo::ConstVariableRefs{
+        std::cref(DISPLACEMENT_X), std::cref(DISPLACEMENT_Y), std::cref(WATER_PRESSURE)};
+    auto& r_model_part = ModelSetupUtilities::CreateModelPartWithASingle2D3NElement(model, nodal_variables);
+
+    const auto& r_geometry = r_model_part.Elements().front().GetGeometry();
+    const auto  dofs =
+        Geo::DofUtilities::ExtractUPwDofsFromNodes(r_geometry, r_geometry.WorkingSpaceDimension());
+
+    KRATOS_EXPECT_EQ(dofs.size(), r_model_part.NumberOfNodes() * nodal_variables.size());
+    ExpectDofsDontContainAnyNullptrs(dofs);
+    ExpectDofsHaveThisVariable({dofs[0], dofs[2], dofs[4]}, DISPLACEMENT_X);
+    ExpectDofsHaveThisVariable({dofs[1], dofs[3], dofs[5]}, DISPLACEMENT_Y);
+    ExpectDofsHaveThisVariable({dofs.begin() + 6, dofs.end()}, WATER_PRESSURE);
+    ExpectDofsHaveThisNodeId({dofs[0], dofs[1], dofs[6]}, 1);
+    ExpectDofsHaveThisNodeId({dofs[2], dofs[3], dofs[7]}, 2);
+    ExpectDofsHaveThisNodeId({dofs[4], dofs[5], dofs[8]}, 3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UDofsPrecedePwDofsWhenExtractingUPwDofsFromNondiffOrder3DElement, KratosGeoMechanicsFastSuite)
+{
+    auto       model = Model{};
+    const auto nodal_variables =
+        Geo::ConstVariableRefs{std::cref(DISPLACEMENT_X), std::cref(DISPLACEMENT_Y),
+                               std::cref(DISPLACEMENT_Z), std::cref(WATER_PRESSURE)};
+    auto& r_model_part = ModelSetupUtilities::CreateModelPartWithASingle3D4NElement(model, nodal_variables);
+
+    const auto& r_geometry = r_model_part.Elements().front().GetGeometry();
+    const auto  dofs =
+        Geo::DofUtilities::ExtractUPwDofsFromNodes(r_geometry, r_geometry.WorkingSpaceDimension());
+
+    KRATOS_EXPECT_EQ(dofs.size(), r_model_part.NumberOfNodes() * nodal_variables.size());
+    ExpectDofsDontContainAnyNullptrs(dofs);
+    ExpectDofsHaveThisVariable({dofs[0], dofs[3], dofs[6], dofs[9]}, DISPLACEMENT_X);
+    ExpectDofsHaveThisVariable({dofs[1], dofs[4], dofs[7], dofs[10]}, DISPLACEMENT_Y);
+    ExpectDofsHaveThisVariable({dofs[2], dofs[5], dofs[8], dofs[11]}, DISPLACEMENT_Z);
+    ExpectDofsHaveThisVariable({dofs.begin() + 12, dofs.end()}, WATER_PRESSURE);
+    ExpectDofsHaveThisNodeId({dofs[0], dofs[1], dofs[2], dofs[12]}, 1);
+    ExpectDofsHaveThisNodeId({dofs[3], dofs[4], dofs[5], dofs[13]}, 2);
+    ExpectDofsHaveThisNodeId({dofs[6], dofs[7], dofs[8], dofs[14]}, 3);
+    ExpectDofsHaveThisNodeId({dofs[9], dofs[10], dofs[11], dofs[15]}, 4);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(UDofsPrecedePwDofsWhenExtractingUPwDofsFromDiffOrder2DElement, KratosGeoMechanicsFastSuite)
+{
+    auto model = Model{};
+    auto& r_model_part = ModelSetupUtilities::CreateModelPartWithASingle2D6NUPwDiffOrderElement(model);
+
+    const auto second_order_nodes = r_model_part.Elements().front().GetGeometry();
+    const auto first_order_nodes =
```

Comment:
In the time integration schemes we introduces first and second order variables, because we have Pw and T type variables in the same scheme. The UPw element always has U and Pw D.O.F so maybe refer to u_nodes and pw_nodes.

Reply:
Renamed the variables to `displacement_nodes` and `water_pressure_nodes`, respectively.